crates/bevy_reflect/src/serde/mod.rs

```diff
@@ -104,4 +112,138 @@ mod tests {
+            where
+                D: Deserializer<'de>,
+            {
+                struct PetVisitor<'a> {
+                    registry: &'a TypeRegistry,
+                }
+                impl<'a, 'de> Visitor<'de> for PetVisitor<'a> {
+                    type Value = Pet;
+
+                    fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+                        write!(formatter, "a pet tuple struct")
+                    }
+
+                    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+                    where
+                        A: SeqAccess<'de>,
+                    {
+                        let kind = seq.next_element::<AnimalType>()?.unwrap();
+                        match kind {
+                            AnimalType::Cat => {
+                                let cat = seq
+                                    .next_element_seed(TypedReflectDeserializer::of::<Cat>(
+                                        self.registry,
+                                    ))?
+                                    .unwrap()
+                                    .take::<DynamicStruct>()
+                                    .unwrap();
+                                Ok(Pet(kind, cat))
+                            }
+                            AnimalType::Dog => {
+                                let dog = seq
+                                    .next_element_seed(TypedReflectDeserializer::of::<Dog>(
+                                        self.registry,
+                                    ))?
+                                    .unwrap()
+                                    .take::<DynamicStruct>()
+                                    .unwrap();
+                                Ok(Pet(kind, dog))
+                            }
+                        }
+                    }
+                }
+
+                deserializer.deserialize_tuple_struct("Pet", 1, PetVisitor { registry })
+            }
+        }
+
+        let mut registry = TypeRegistry::default();
+        registry.register::<Pet>();
+        registry.register::<AnimalType>();
+        registry.register::<Dog>();
+        registry.register::<DogName>();
+        registry.register::<Cat>();
+        registry.register::<CatName>();
+
+        let pet = Pet(
+            AnimalType::Cat,
+            Cat {
+                name: CatName::Fluffy,
+            }
+            .clone_dynamic(),
+        );
+
+        let serializer = ReflectSerializer::new(&pet, &registry);
+        let serialized = ron::ser::to_string(&serializer).unwrap();
+
+        let expected = r#"{"bevy_reflect::serde::tests::should_deserialize_using_deserialize_reflect::Pet":(Cat,(name:Fluffy))}"#;
+
+        assert_eq!(expected, serialized);
+
+        let mut deserializer = ron::de::Deserializer::from_str(&serialized).unwrap();
+        let reflect_deserializer = UntypedReflectDeserializer::new(&registry);
```

Comment:
This may be obvious/stupid but why not use `TypedReflectDeserializer` here, instead of `UntypedReflectDeserializer`? Or do both work?

Reply:
Both work. `UntypedReflectDeserializer` actually calls `TypedReflectDeserializer` internally, so I suppose it's one way of testing that both work. 

Although, in this particular case I think I just forgot that I added `TypedReflectDeserializer::of` haha

Reply:
Maybe try to explicitly test both then? Or just prefer testing the new `::of()` for coverage?

Reply:
I don't think I need to switch this anymore. The `of` method actually won't work for this test since we're deserializing the map rather than the value directly.